### PR TITLE
[FEATURE] Support list-table directive

### LINKED ISSUE #890: Support "list-table" directive
### Feature request

The `list-table` directive is currently not supported. It would be nice to have this also in guides:
https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table

### PR DESCRIPTION
**Pull request by linawolf**
https://sublime-and-sphinx-guide.readthedocs.io/en/latest/tables.html

resolves #890

@jaapio can we still backport features?

### CODE REVIEW

**Comment by jaapio** [comment]
Yes for sure as they are not breaking changes we can

File: packages/guides-restructured-text/src/RestructuredText/Parser/Productions/ListRule.php [hunk 1]
```diff
@@ -59,7 +58,6 @@ final class ListRule implements Rule
 
     public function __construct(
         private readonly RuleContainer $productions,
-        private readonly LoggerInterface $logger,
```
Comment (jaapio) [inline packages/guides-restructured-text/src/RestructuredText/Parser/Productions/ListRule.php:62]: I thinks we should not remove this right now, it's not part of the change, and I see removing an argument as a breaking change. Also because a change in the tests is required. If this leads to an unused property, we could transform it to a normal parameter, and make it optional. 
Add a deprecation and remove it in v2.

File: packages/guides-restructured-text/tests/unit/Parser/Productions/ListRuleTest.php [hunk 2]
```diff
@@ -26,7 +25,7 @@ final class ListRuleTest extends RuleTestCase
     protected function setUp(): void
     {
         $ruleContainer = $this->givenCollectAllRuleContainer();
-        $this->rule = new ListRule($ruleContainer, new NullLogger());
+        $this->rule = new ListRule($ruleContainer);
```
Comment (jaapio) [inline packages/guides-restructured-text/tests/unit/Parser/Productions/ListRuleTest.php:28]: please leave the tests untouched, whenever you need to change a unittest that might indicate a breaking change :-)

File: packages/guides/src/Compiler/NodeTransformers/ListNodeTransformer.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public function leaveNode(Node $node, CompilerContext $compilerContext): Node|null
+    {
+        assert($node instanceof ListNode);
+        foreach ($node->getChildren() as $listItemNode) {
+            assert($listItemNode instanceof ListItemNode);
+            if (!empty($listItemNode->getChildren())) {
+                continue;
+            }
+
+                        $this->logger->warning('List item without content', $compilerContext->getLoggerInformation());
```
Comment (jaapio) [inline packages/guides/src/Compiler/NodeTransformers/ListNodeTransformer.php:47]: Indending seems to be of here.

**Review by jaapio — COMMENTED** [review]
Just some minor changes, rest looks good. Thanks for adding this!

File: packages/guides-restructured-text/src/RestructuredText/Parser/Productions/ListRule.php (hunk 1 above)
Comment (linawolf) [reply packages/guides-restructured-text/src/RestructuredText/Parser/Productions/ListRule.php]: I just added it to the phpstan baseline. we might need it again , there is no harm in leaving it in

File: packages/guides-restructured-text/tests/unit/Parser/Productions/ListRuleTest.php (hunk 2 above)
Comment (linawolf) [reply packages/guides-restructured-text/tests/unit/Parser/Productions/ListRuleTest.php]: done

File: packages/guides/src/Compiler/NodeTransformers/ListNodeTransformer.php (hunk 3 above)
Comment (linawolf) [reply packages/guides/src/Compiler/NodeTransformers/ListNodeTransformer.php]: done

**Review by jaapio — APPROVED** [review]
